sessionctx/varsutil/varsutil.go

```diff
@@ -154,26 +157,30 @@ func tidbOptPositiveInt(opt string, defaultVal int) int {
 	return val
 }
 
-func parseTimeZone(s string) *time.Location {
+func parseTimeZone(s string) (*time.Location, error) {
 	if s == "SYSTEM" {
 		// TODO: Support global time_zone variable, it should be set to global time_zone value.
-		return time.Local
+		return time.Local, nil
 	}
 
 	loc, err := time.LoadLocation(s)
 	if err == nil {
-		return loc
+		return loc, nil
 	}
 
 	// The value can be given as a string indicating an offset from UTC, such as '+10:00' or '-6:00'.
 	if strings.HasPrefix(s, "+") || strings.HasPrefix(s, "-") {
 		d, err := types.ParseDuration(s[1:], 0)
 		if err == nil {
-			return time.FixedZone("UTC", int(d.Duration/time.Second))
+			ofst := int(d.Duration / time.Second)
+			if s[0] == '-' {
+				ofst = -ofst
+			}
+			return time.FixedZone("UTC", ofst), nil
 		}
 	}
 
-	return nil
+	return nil, errors.Errorf("Unknown or incorrect time zone: %s", s)
```

Comment:
MySQL use 1298 for incorrect time zone error.

mysql> SET GLOBAL time_zone = '+xxxxx';
ERROR 1298 (HY000): Unknown or incorrect time zone: '+xxxxx'
mysql> 

➜  mysql git:(master) ✗ grep 1298 -r ./
.//errcode.go:	ErrUnknownTimeZone                                              = 1298
➜  mysql git:(master) ✗